docs/DeveloperGuide.adoc

```diff
@@ -153,94 +149,120 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
-// tag::undoredo[]
-=== [Proposed] Undo/Redo feature
-==== Proposed Implementation
+// tag::moduleview[]
+=== Module View Feature
 
-The undo/redo mechanism is facilitated by `VersionedAddressBook`.
-It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
-Additionally, it implements the following operations:
+The module view feature allows users to view a module to access extra information about the module. The user can view (and unview) a module using the `view` (and `back`) commands.
 
-* `VersionedAddressBook#commit()` -- Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` -- Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` -- Restores a previously undone address book state from its history.
+==== Implementation
+The module view feature is facilitated by `displayedModule` in `ModelManager`. The following describes how the `displayedModule` is modified and read to the `Ui`.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+. The user enters a view command in the `Ui`.
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+. `LogicManager` parses the user input, constructs and executes the `ViewCommand`.
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+. The `ViewCommand` attempts to find the matching module from the list of Tracked Modules, else it attempts to find the matching module from the list of Archived Modules, else it throws an exception.
 
-image::UndoRedoState0.png[]
+. The `ViewCommand` sets the `displayedModule` in the `Model` and returns the `CommandResult` to the `LogicManager`.
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+. The `LogicManager` returns the `CommandResult` to the `Ui`.
 
-image::UndoRedoState1.png[]
+. The `Ui` gets the `displayedModule` from `LogicManager` and updates the Ui to display the module.
 
-Step 3. The user executes `add n/David ...` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+The following sequence diagram shows how the module view feature is executed.
 
-image::UndoRedoState2.png[]
+image::ModuleViewSequenceDiagram.png[]
```

Comment:
Hello! I don't think you can write something like "create and execute"? Unless your method is called "create and execute"? Also, where's the "find matching module" diagram that you mentioned in your ref? 😺 😺

Reply:
The actual implementation is abstracted out so as to keep the diagram simpler